server/schedule/operator/create_operator.go

```diff
@@ -282,3 +290,24 @@ func CreateLeaveJointStateOperator(desc string, ci ClusterInformer, origin *core
 	b.execChangePeerV2(false, true)
 	return NewOperator(b.desc, brief, b.regionID, b.regionEpoch, kind, origin.GetApproximateSize(), b.steps...), nil
 }
+
+// CreateWitnessPeerOperator creates an operator that set a follower or learner peer with witness
+func CreateWitnessPeerOperator(desc string, ci ClusterInformer, region *core.RegionInfo, peer *metapb.Peer) (*Operator, error) {
+	brief := fmt.Sprintf("create witness: region %v peer %v on store %v", region.GetID(), peer.Id, peer.StoreId)
+	return NewOperator(desc, brief, region.GetID(), region.GetRegionEpoch(), OpRegion, region.GetApproximateSize(), BecomeWitness{StoreID: peer.StoreId, PeerID: peer.Id}), nil
+}
+
+// CreateNonWitnessPeerOperator creates an operator that set a peer with non-witness
+func CreateNonWitnessPeerOperator(desc string, ci ClusterInformer, region *core.RegionInfo, peer *metapb.Peer) (*Operator, error) {
```

Comment:
Can we change a peer from witness to non-witness directly?

Reply:
Yes, we can